app/controllers/walks_controller.rb

```diff
@@ -33,7 +35,8 @@ def update
     @walk.guardian_id ||= params[:guardian_id]
 
     if @walk.save
-      redirect_to user_walk_path(@walk)
+      @walk.chatroom = Chatroom.create(topic: @walk.id)
+      redirect_to @walk.chatroom
```

Comment:
I checked the commit history, and between commits 5 and 6 it looks like you reverted to this model instead of the callback method.